crates/store/src/state.rs

```diff
@@ -96,17 +96,114 @@ pub struct TransactionInputs {
     pub found_unauthenticated_notes: BTreeSet<NoteId>,
 }
 
+#[derive(Debug)]
+pub struct Blockchain(Mmr);
+
+impl Blockchain {
+    pub fn get(&self) -> &Mmr {
+        &self.0
+    }
+    pub fn get_mut(&mut self) -> &mut Mmr {
+        &mut self.0
+    }
+
+    pub fn get_latest_block_number(&self) -> BlockNumber {
```

Comment:
nit: I think `chain_tip()` be a more concrete name here (aligns more with other structures AFAICT)